### PR TITLE
Vcpkg sample

### LINKED ISSUE #3669: Investigate Vcpkg sample failure in nightly pipeline

### PR DESCRIPTION
**Pull request by gearama**
closes #3669 
Fix vcpkg sample. 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: eng/pipelines/templates/jobs/live.tests.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ parameters:
   default: sdk/*/*/*cov_xml.xml
 - name: TimeoutInMinutes
   type: number
-  default: 60
+  default: 120
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/live.tests.yml:22]: Do we really need to move the default timeout? Are the samples adding that much time to the pipeline runs?

Comment (gearama) [reply eng/pipelines/templates/jobs/live.tests.yml:22]: Yes , actually they are , a couple of the jobs run way over one hour causing the pipeline to fail.

Comment (weshaggard) [reply eng/pipelines/templates/jobs/live.tests.yml:22]: OK I guess as a team you guys should decide if you want to be running all the samples if they add that much time or perhaps work on the performance of them.

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 2]
```diff
@@ -43,9 +44,9 @@ int main()
   {
     std::cout << "Creating Keyvault Clients" << std::endl;
     // keyvault
-    KeyClient keyClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
-    SecretClient secretClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
-    CertificateClient certificateClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
+    KeyClient keyClient(smokeUrl, credential);
+    SecretClient secretClient(smokeUrl, credential);
+    CertificateClient certificateClient(smokeUrl, credential);
```
Comment (antkmsft) [inline samples/integration/vcpkg-all-smoke/src/main.cpp:49]: We are using `smokeUrl` for attestation client and for KeyVault URLs. How will this work?

Comment (gearama) [reply samples/integration/vcpkg-all-smoke/src/main.cpp:49]: it's not supposed to work. it's supposed to test that we can instantiate the clients. not that the clients conenct to random services. this test is purely to test that our vcpkg packages don't blow each other up.

Comment (LarryOsterman) [reply samples/integration/vcpkg-all-smoke/src/main.cpp:49]: If the sample isn't supposed to work, then it's not a "sample" (by definition, samples are examples for customers to use to determine how to use our SDKs).

More importantly: If the sample isn't supposed to work, then why are we executing the sample? Isn't it sufficient to ensure that the code compiles?

**Review by LarryOsterman — CHANGES_REQUESTED** [review]

File: samples/integration/vcpkg-all-smoke/src/main.cpp (hunk 2 above)
Comment (gearama) [reply samples/integration/vcpkg-all-smoke/src/main.cpp:49]: THIS IS NOT FOR CUSTOMER USE , it's for internal vcpkg pipeline to validate that the packages load properly.

Comment (LarryOsterman) [reply samples/integration/vcpkg-all-smoke/src/main.cpp:49]: If this is not for customer use, 
1. It should not be in a directory named "samples"
2. There should be SCREAMINGLY LARGE AND ANNOYING TEXT explaining why customers should ignore it.
3. If it's not supposed to actually do anything meaningful, we shouldn't be expecting it to execute, ever.

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 3]
```diff
@@ -69,10 +70,32 @@ int main()
...
-    std::unique_ptr<AttestationAdministrationClient> attestationAdminClient(
-        AttestationAdministrationClientFactory::Create(std::getenv("ATTESTATION_AAD_URL"), credential));
+
+    // we define what the behavior is in certain scenarios.
+    // exception in this case is expected behavior.
+    try
+    {
+      std::unique_ptr<AttestationAdministrationClient> attestationAdminClient(
+          AttestationAdministrationClientFactory::Create(smokeUrl, credential));
+      return 1;
```
[the remark this answers is not in the document]
Comment (gearama) [reply samples/integration/vcpkg-all-smoke/src/main.cpp]: done, even though this is not supposed to touch actual resources, if asked why , i'll point them in your direction :)

**Comment by gearama** [comment]
@LarryOsterman updated PR

File: samples/integration/vcpkg-all-smoke/src/main.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (LarryOsterman) [reply samples/integration/vcpkg-all-smoke/src/main.cpp]: Unfortunately, it's not possible to instantiate an attestation client without touching some actual resources.

**Review by LarryOsterman — APPROVED** [review]

**Pull request merged**